### DISCUSSION TITLE
How do I concat two columns?

### DISCUSSION THREAD

**Discussion by G1vas** [saulpw/visidata#2642] [Q&A] [answered]
Hello,

I am having a hard time finding out how do I concat two columns? For example:
Column A:
A|
-- |
1|
2|
3|

Column B:
B|
-- |
4|
5|
6|

concated column:
C|
-- |
1|
2|
3|
4|
5|
6|

Is this possible?

Thank you.

**Comment by anjakefala** [comment]
Yes! This is known as appending in VisiData: https://www.visidata.org/docs/join/


    Open the datasets with VisiData.
    Press `Shift+S` to open the Sheets sheet.
    Use `s` or `t` to select the sheets to concate.
    Type `&` and select `append` to concatenate the selected datasets.

**Reply by G1vas** [reply]
I forgot to mention that both of the columns are in the same sheet. I've already managed to concat two columns from two different sheets, but I do not understand how can I join two columns that are in the same sheet:(

**Reply by anjakefala** [reply]
There might be a better way, but I would `dup-sheet` (`"`), and then treat them as key columns on two different sheets.

**Reply by G1vas** [reply]
Thanks for your help. I will keep this question unanswered for a few days, maybe someone else has an even better solution:)

**Comment by asteroidb612** [comment, marked as answer]
I might have another way! 

Given a sheet with two columns, `Shift+M` will "Melt" them into one column. 

Here is some more [documentation](https://jsvine.github.io/intro-to-visidata/intermediate/reshaping-data/#melting-datasets) on reshaping data, including melts and pivots.

**Reply by G1vas** [reply]
This is a very good solution, thank you. It works better than duping sheets for me since I work with very big data sets.

**Reply by midichef** [reply]
That's a creative application of melt! I hadn't thought of that.

**Comment by midichef** [comment]
Another solution you can try is using the system cut buffer. Copy the column into it usying `syscopy-cells`:
`gs` `gzY`
then move to the destination column, add a row, and paste the copied data at the end using `syspaste-cells`:
`End` `a` `zP` (or `MiddleBtnUp`)
